ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java

```diff
@@ -1130,4 +1116,16 @@ void cancel() {
       cancelled = true;
     }
   }
+
+  /**
+   * An action to take on a single {@link IndexEntry}
+   */
+  @FunctionalInterface
+  private interface IndexEntryAction {
```

Comment:
I added a new interface instead of using `Consumer<IndexEntry>` or `ThrowingConsumer<IndexEntry>` since these some of these methods need to throw `StoreException` specifically unless many method signatures are changed